libs/@local/hash-graph-types/rust/src/knowledge/entity/provenance.rs

```diff
@@ -7,9 +7,167 @@ use bytes::BytesMut;
 use postgres_types::{FromSql, IsNull, Json, ToSql, Type};
 use serde::{Deserialize, Serialize};
 use temporal_versioning::{DecisionTime, Timestamp, TransactionTime};
+use time::OffsetDateTime;
+use url::Url;
 
 use crate::account::{CreatedById, EditionArchivedById, EditionCreatedById};
 
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[cfg_attr(feature = "utoipa", derive(utoipa::ToSchema))]
+#[serde(deny_unknown_fields, rename_all = "camelCase")]
+pub struct Tool {
+    /// The name of the tool.
+    pub name: String,
+
+    /// The tool version, in whatever format the tool natively provides.
+    #[cfg_attr(feature = "utoipa", schema(nullable = false))]
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub version: Option<String>,
+
+    /// The tool version in the format specified by Semantic Versioning 2.0.
+    #[cfg_attr(feature = "utoipa", schema(nullable = false, value_type = String))]
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub semantic_version: Option<semver::Version>,
+
+    /// The organization or company that produced the tool.
+    #[cfg_attr(feature = "utoipa", schema(nullable = false))]
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    pub organization: Option<String>,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[cfg_attr(feature = "utoipa", derive(utoipa::ToSchema))]
+#[serde(deny_unknown_fields, rename_all = "camelCase")]
+pub struct Artifact {
```

Comment:
I would be tempted to leave this out until we know how exactly we will use it... we can discuss Monday

Reply:
Removed in [`002ddaf` (#4306)](https://github.com/hashintel/hash/pull/4306/commits/002ddaf85c18c4f1284b156e56c487fd50cb1b61)